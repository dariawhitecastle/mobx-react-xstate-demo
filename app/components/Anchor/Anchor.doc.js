import { describe, PropTypes } from 'react-desc'

import { colorPropType, genericProps } from '../../utils/propTypes'
import { themeDocUtils } from '../../utils/themeDocUtils'
export const doc = Anchor => {
  const DocumentedAnchor = describe(Anchor)
    .description('A text link.')
    .details(
      `We have a separate component from the browser
base so we can style it. You can either set the icon and/or label properties
or just use children.`,
    )
    .usage("import { Anchor } from 'components/Anchor';\n<Anchor href={location} label='Label' />")
    .intrinsicElement('a')

  DocumentedAnchor.propTypes = {
    ...genericProps,
    a11yTitle: PropTypes.string.description('Custom title to be used by screen readers.'),
    color: colorPropType.description('Label color and icon color, if not specified on the icon.'),
    href: PropTypes.string.description('Hyperlink reference to place in the anchor.'),
    icon: PropTypes.element.description('Icon element to place in the anchor.'),
    label: PropTypes.node.description('Label text to place in the anchor.'),
    onClick: PropTypes.func.description(
      `Click handler. It can be used, for example,
        to add analytics and track who clicked in the anchor.`,
    ),
    reverse: PropTypes.bool
      .description(
        `Whether an icon and label should be reversed so that the
        icon is at the end of the anchor.`,
      )
      .defaultValue(false),
    size: PropTypes.oneOfType([
      PropTypes.oneOf(['xsmall', 'small', 'medium', 'large', 'xlarge', 'xxlarge']),
      PropTypes.string,
    ]).description(
      `The font size is typically driven by the components containing
this component. But, it can be adjusted directly via this size property, typically
when it is not contained in a 'Heading', 'Paragraph', or 'Text'.`,
    ),
    as: PropTypes.string.description(`The DOM tag to use for the element.`),
  }

  return DocumentedAnchor
}

export const themeDoc = {
  'anchor.color': {
    description: 'The color of the label text and icon strokes.',
    type: 'string | { dark: string, light: string }',
    defaultValue: "{ light: '#1D67E3', dark: '#6194EB' }",
  },
  'anchor.fontWeight': {
    description: 'The font weight of the label.',
    type: 'number',
    defaultValue: 600,
  },
  'anchor.textDecoration': {
    description:
      'The text decoration of the label. Refer to [MDN](https://developer.mozilla.org/en-US/docs/Web/CSS/text-decoration) for possible values.',
    type: 'string',
    defaultValue: 'none',
  },
  'anchor.hover.fontWeight': {
    description: 'The font weight of the label when hovering.',
    type: 'number',
    defaultValue: undefined,
  },
  'anchor.hover.textDecoration': {
    description:
      'The text decoration of the label when hovering. Refer to [MDN](https://developer.mozilla.org/en-US/docs/Web/CSS/text-decoration) for possible values.',
    type: 'string',
    defaultValue: 'underline',
  },
  'anchor.hover.extend': {
    description: 'Any additional style for the Anchor when hovering.',
    type: 'string | (props) => {}',
    defaultValue: undefined,
  },
  'anchor.extend': {
    description: 'Any additional style for the Anchor.',
    type: 'string | (props) => {}',
    defaultValue: undefined,
  },
  ...themeDocUtils.focusStyle,
}
