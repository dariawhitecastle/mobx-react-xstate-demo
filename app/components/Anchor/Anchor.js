/* eslint-disable react/prop-types */
/**
 *
 * Anchor
 *
 */

import React from 'react'
import { Anchor } from 'grommet'
import { doc } from './Anchor.doc'
import helpers from '../../utils/helpers'

function createAnchor({
  a11yTitle,
  children,
  color,
  disabled,
  forwardRef,
  href,
  icon,
  focus,
  label,
  onClick,
  reverse,
  theme,
  size,
  ...rest
}) {
  return (
    <Anchor
      a11yTitle={a11yTitle}
      color={color}
      disabled={disabled}
      forwardRef={forwardRef}
      href={href}
      icon={icon}
      focus={focus}
      label={label}
      onClick={onClick}
      reverse={reverse}
      theme={theme}
      size={size}
      {...rest}
    >
      {children}
    </Anchor>
  )
}

export default helpers.createWithDoc({
  docFunction: doc,
  component: createAnchor,
})
