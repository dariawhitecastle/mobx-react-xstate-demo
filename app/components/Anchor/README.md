<!-- ! This is a generated file. To make changes, edit <Component>.doc.js ! -->
## createAnchor
A text link.

We have a separate component from the browser
base so we can style it. You can either set the icon and/or label properties
or just use children.

## Usage

```javascript
import { Anchor } from 'components/Anchor';
<Anchor href={location} label='Label' />
```

## Properties

**a11yTitle**

Custom title to be used by screen readers.

```
string
```

**alignSelf**

How to align along the cross axis when contained in
      a Box or along the column axis when contained in a Grid.

```
start
center
end
stretch
```

**gridArea**

The name of the area to place
    this inside a parent Grid.

```
string
```

**margin**

The amount of margin around the component. An object can
      be specified to distinguish horizontal margin, vertical margin, and
      margin on a particular side.

```
none
xxsmall
xsmall
small
medium
large
xlarge
{
  bottom: 
    xxsmall
    xsmall
    small
    medium
    large
    xlarge
    string,
  horizontal: 
    xxsmall
    xsmall
    small
    medium
    large
    xlarge
    string,
  left: 
    xxsmall
    xsmall
    small
    medium
    large
    xlarge
    string,
  right: 
    xxsmall
    xsmall
    small
    medium
    large
    xlarge
    string,
  top: 
    xxsmall
    xsmall
    small
    medium
    large
    xlarge
    string,
  vertical: 
    xxsmall
    xsmall
    small
    medium
    large
    xlarge
    string
}
string
```

**color**

Label color and icon color, if not specified on the icon.

```
string
{
  dark: string,
  light: string
}
```

**href**

Hyperlink reference to place in the anchor.

```
string
```

**icon**

Icon element to place in the anchor.

```
element
```

**label**

Label text to place in the anchor.

```
node
```

**onClick**

Click handler. It can be used, for example,
        to add analytics and track who clicked in the anchor.

```
function
```

**reverse**

Whether an icon and label should be reversed so that the
        icon is at the end of the anchor.

```
boolean
```

**size**

The font size is typically driven by the components containing
this component. But, it can be adjusted directly via this size property, typically
when it is not contained in a 'Heading', 'Paragraph', or 'Text'.

```
xsmall
small
medium
large
xlarge
xxlarge
string
```

**as**

The DOM tag to use for the element.

```
string
```
  
## Intrinsic element

```
a
```
## Theme
  
**anchor.color**

The color of the label text and icon strokes. Expects `string | { dark: string, light: string }`.

Defaults to

```
{ light: '#1D67E3', dark: '#6194EB' }
```

**anchor.fontWeight**

The font weight of the label. Expects `number`.

Defaults to

```
600
```

**anchor.textDecoration**

The text decoration of the label. Refer to [MDN](https://developer.mozilla.org/en-US/docs/Web/CSS/text-decoration) for possible values. Expects `string`.

Defaults to

```
none
```

**anchor.hover.fontWeight**

The font weight of the label when hovering. Expects `number`.

Defaults to

```
undefined
```

**anchor.hover.textDecoration**

The text decoration of the label when hovering. Refer to [MDN](https://developer.mozilla.org/en-US/docs/Web/CSS/text-decoration) for possible values. Expects `string`.

Defaults to

```
underline
```

**anchor.hover.extend**

Any additional style for the Anchor when hovering. Expects `string | (props) => {}`.

Defaults to

```
undefined
```

**anchor.extend**

Any additional style for the Anchor. Expects `string | (props) => {}`.

Defaults to

```
undefined
```

**global.focus.border.color**

The color around the component when in focus. Expects `string | { dark: string, light: string }`.

Defaults to

```
focus
```
