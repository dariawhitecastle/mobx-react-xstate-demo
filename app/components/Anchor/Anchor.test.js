/**
 *
 * Tests for Anchor
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react'
import { fireEvent, render } from '@testing-library/react'
import 'jest-styled-components'
import { Add } from 'grommet-icons'

import Anchor from './Anchor'

const children = <h1>Test</h1>
const href = 'https://cuttlesoft.com'

const additionalProps = {
  a11yTitle: 'string',
  color: 'blue',
  disabled: false,
  icon: <Add />,
  label: 'Clickme',
  onClick: () => {},
  reverse: false,
  size: 'small',
}

const renderComponent = props =>
  render(
    <Anchor href={href} {...props}>
      {children}
    </Anchor>,
  )

test('it should handle click events', () => {
  const onClickSpy = jest.fn()
  const { container } = renderComponent({ onClick: onClickSpy })
  fireEvent.click(container.querySelector('a'))
  expect(onClickSpy).toHaveBeenCalled()
})

test('it renders and matches snapshot', () => {
  const { container } = renderComponent(additionalProps)
  expect(container).toMatchSnapshot()
})
