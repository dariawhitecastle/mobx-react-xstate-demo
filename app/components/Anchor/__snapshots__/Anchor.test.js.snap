// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders and matches snapshot 1`] = `
.c2 {
  display: inline-block;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  width: 24px;
  height: 24px;
  fill: blue;
  stroke: blue;
}

.c2 g {
  fill: inherit;
  stroke: inherit;
}

.c2 *:not([stroke])[fill="none"] {
  stroke-width: 0;
}

.c2 *[stroke*="#"],
.c2 *[STROKE*="#"] {
  stroke: inherit;
  fill: none;
}

.c2 *[fill-rule],
.c2 *[FILL-RULE],
.c2 *[fill*="#"],
.c2 *[FILL*="#"] {
  fill: inherit;
  stroke: none;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  outline: none;
  max-width: 100%;
  margin: 0px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  padding: 0px;
}

.c3 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  width: 12px;
}

.c0 {
  box-sizing: border-box;
  font-size: 14px;
  line-height: 20px;
  color: blue;
  font-weight: 600;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  outline: none;
}

.c0:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

@media only screen and (max-width:768px) {
  .c1 {
    margin: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c1 {
    padding: 0px;
  }
}

<div>
  <a
    aria-label="string"
    class="c0"
    href="https://cuttlesoft.com"
  >
    <span
      class="c1"
      style="display: inline-flex;"
    >
      <svg
        aria-label="Add"
        class="c2"
        viewBox="0 0 24 24"
      >
        <path
          d="M12,22 L12,2 M2,12 L22,12"
          fill="none"
          stroke="#000"
          stroke-width="2"
        />
      </svg>
      <div
        class="c3"
      />
      Clickme
    </span>
  </a>
</div>
`;
